brainscore_core/submission/database.py

```diff
@@ -0,0 +1,163 @@
+        model_entry.save()
+    return model_entry
+
+
+def benchmarkinstance_from_benchmark(benchmark: Benchmark) -> BenchmarkInstance:
+    benchmark_identifier = benchmark.identifier
+    benchmark_type, created = BenchmarkType.get_or_create(identifier=benchmark_identifier, defaults=dict(order=999))
+    if created:
+        # store parent
+        try:
+            parent = BenchmarkType.get(identifier=benchmark.parent)
+            benchmark_type.parent = parent
+            benchmark_type.save()
+        except DoesNotExist:
+            logger.warning(f'Could not connect benchmark {benchmark_identifier} to parent {benchmark.parent} '
+                           f'since parent does not exist')
+        # store reference
+        if hasattr(benchmark, 'bibtex') and benchmark.bibtex is not None:
+            bibtex_string = benchmark.bibtex
+            ref = reference_from_bibtex(bibtex_string)
+            if ref:
+                benchmark_type.reference = ref
+                benchmark_type.save()
+
+    # process instance
+    bench_inst, created = BenchmarkInstance.get_or_create(benchmark=benchmark_type, version=benchmark.version)
+    if created:
+        # the version has changed and the benchmark instance was not yet in the database
+        ceiling = benchmark.ceiling
+        bench_inst.ceiling = ceiling.item()
+        bench_inst.ceiling_error = _retrieve_score_error(ceiling)
+        bench_inst.save()
+    return bench_inst
+
+
+def reference_from_bibtex(bibtex_string: str) -> Union[Reference, None]:
+    def parse_bib(bibtex_str):
+        bib_parser = bibtex.Parser()
+        entry = bib_parser.parse_string(bibtex_str)
+        entry = entry.entries
+        assert len(entry) == 1
+        entry = list(entry.values())[0]
+        return entry
+
+    try:
+        entry = parse_bib(bibtex_string)
+        ref, created = Reference.get_or_create(url=entry.fields['url'],
+                                               defaults={'bibtex': bibtex_string,
+                                                         'author': entry.persons["author"][0].last()[0],
+                                                         'year': entry.fields['year']})
+        return ref
+    except Exception:
+        logger.exception('Could not load reference from bibtex string')
+        return None
+
+
+def update_score(score: ScoreObject, entry: Score):
+    if 'ceiling' not in score.attrs:  # many engineering benchmarks do not have a primate ceiling
+        # only store raw (unceiled) value
+        entry.score_raw = _retrieve_score_center(score)
+    else:  # score has a ceiling. Store ceiled as well as raw value
+        score_raw = _retrieve_score_center(score.raw)
+        entry.score_raw = score_raw
+        entry.score_ceiled = _retrieve_score_center(score)
+    entry.error = _retrieve_score_error(score)
+    print(f"updating raw score: {entry.score_raw}, ceiled score: {entry.score_ceiled}, error: {entry.error}")
+    print(entry.save())
```

Comment:
```suggestion
    logger.debug(f"updating raw score: {entry.score_raw}, ceiled score: {entry.score_ceiled}, error: {entry.error}")
    entry.save()
```
`save()` doesn't return anything AFAIK